src/devices/Mbi5027/Mbi5027.cs

```diff
@@ -0,0 +1,186 @@
+                 1   1   1   1   1
+            SDI ‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾|__
+                 1   1   1   1   0
+            LE  _________________↑‾|
+                 0   0   0   0   1
+
+            **Phase 2: Signal to detect error status
+                  1   2   3
+            CLK _↑‾|_↑‾|_↑‾|
+            OE  ___________
+                 0   0   0
+
+            **Phase 3: Read status for each output (normal or error codes)
+                  1   2   3   n
+            CLK _↑‾|_↑‾|_↑‾|_↑‾|
+
+            OE  ‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾‾
+                 1   1   1   1
+            SDO  ?   ?   ?   ?
+            Read error codes from SDO -- starts with BitLength - 1
+            */
+
+            // Phase 1: Load test data
+            // n cycles with OE high
+            for (int i = 0; i < BitLength; i++)
+            {
+                ShiftBit(1);
+            }
+
+            // latch test data
+            Latch();
+
+            // Phase 2: Signal to detect error status
+            // three clock cycles, with OE low
+            GpioController.Write(_pinMapping.OE, 0);
+            for (int i = 0; i < 3; i++)
+            {
+                GpioController.Write(_pinMapping.Clk, 1);
+                GpioController.Write(_pinMapping.Clk, 0);
+            }
+
+            // Phase 3: Read status for each output (normal or error codes)
+            // read error codes from SDO, with OE high
+            GpioController.Write(_pinMapping.OE, 1);
+            for (int i = 0; i < BitLength; i++)
+            {
+                PinValue sdo = GpioController.Read(_pinMapping.Sdo);
+                yield return sdo;
+                GpioController.Write(_pinMapping.Clk, 1);
+                GpioController.Write(_pinMapping.Clk, 0);
+            }
+        }
+
+        /// <summary>
+        /// Enable normal mode.
+        /// Requires use of GPIO controller.
+        /// </summary>
+        public void EnableNormalMode()
+        {
+            if (GpioController is null || _pinMapping.Clk < 0 || _pinMapping.OE < 0 || _pinMapping.LE < 0)
+            {
+                throw new ArgumentNullException($"{nameof(EnableDetectionMode)}: GpioController was not provided or {nameof(_pinMapping.Clk)}, {nameof(_pinMapping.LE)}, or {nameof(_pinMapping.OE)} not mapped to pin");
+            }
+
+            /*  Required timing waveform
+                  1   2   3   4   5
+            CLK _↑‾|_↑‾|_↑‾|_↑‾|_↑‾|
+
+            OE  ‾‾‾|___|‾‾‾‾‾‾‾‾‾‾‾‾
+                 1   0   1   1   1
+            LE  ____________________
+                 0   0   0   0   0
+            */
+
+            ChangeModeSignal(1, 0);
+            ChangeModeSignal(0, 0);
+            ChangeModeSignal(1, 0);
+            ChangeModeSignal(1, 0);
+            ChangeModeSignal(1, 0);
+            GpioController.Write(_pinMapping.OE, 0);
+        }
+
+        private void ChangeModeSignal(PinValue oe, PinValue le)
+        {
+            GpioController.Write(_pinMapping.OE, oe);
```

Comment:
is there any time constraint we should work with? i.e. should there be some delay here?

Reply:
This is a great question. It works as-is. The problem is that we don't have any delays that are at all meaningful. For example, I could add a delay for 1ms, but that would be overkill, by more than 3 orders of magnitude. The chip runs at 25Mhz clock frequency. What's the guidance in that situation?

Reply:
let's leave as is for now, if we find this problematic then we'll figure it out